scripts/create-partner-chart

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+set -e
+
+usage="Usage: make create URL={REQUIRED_URL_TO_CHART} CHART={OPTIONAL_CHART_DIR_NAME}"
+
+# Verify required arguments
+if [[ ! $URL ]]; then
+    echo "Error: Missing required argument URL"
+    echo $usage
+    exit 1
+fi
+
+# Verify URL is valid and references a tgz resource
+if [[ ! $URL =~ https?://.*\.tgz ]]; then
+    echo "Error: Invalid URL form"
+    exit 1
+fi
+curl -fsI $URL &>/dev/null && ok=1
+if [[ ! $ok ]]; then
+    echo "Error: No resource found at $URL"
+    exit 1
+fi
+
+# Try trimming tgz basename to generate a name for chart dir if one wasn't provided
```

Comment:
I'd just force them to supply a name because the url could get weird

Reply:
I check the URL first for these:
1. URL matches the form http://www.name.com/chart-name-v1.0.0.tgz
2. References a resource (checks with curl)
3. Doesn't contain any dots (in case it has extra versions like chart-name-v1.0.0-beta.tgz, etc)

If that fails, a message shows up and the user is forced to supply the name, but I don't expect that to happen as much because we are asking them to use helm repos which usually include GA releases only and standard semantic versioning.

The reason I don't just force them to supply one and rather try creating it first is because they might not look at other charts and just supply a random name with a different case(pascal, camel, underscore, etc), so I at least try to create it for them first. At the end it shouldn't really matter since either case would easily get caught in the PR review.

If you don't agree, I have no problems updating it to force them to supply it. I'm just going based on my experience with partners so far.